src/components/Article-votes.jsx

```diff
@@ -0,0 +1,92 @@
+import { useContext, useEffect, useState } from "react"
+import LoggedInContext from "../contexts/Logged-In-User-Context"
+import { patchArticleById } from "../utils/PATCH-article-votes"
+
+const ArticleVotes = (props) => {
+   
+    const articleVotes = props.articleVotes
+    const setArticleVotes = props.setArticleVotes
+    const currentArticle = props.currentArticle
+    
+    const { loggedInUser, setLoggedInUser } = useContext(LoggedInContext)
+    const [hasVoted, setHasVoted] = useState(false)
+    const [increment, setIncrement] = useState(0)
+    const [isError, setIsError] = useState(false)
+
+    useEffect(() => {
+        setArticleVotes(currentArticle.votes)
+    }, [])
+
+    const handleLike = (event) => {
+        event.preventDefault()
+        const body = {
+            inc_votes : 1,
+        }
+        setIncrement(1)
+        patchArticleById(currentArticle.article_id, body)
+        .then((res) => {
+            setArticleVotes(currentArticle.votes)
+        })
+        .then((res) => {
+            setHasVoted(true)
+        })
+        .catch((err) => {
+            setIsError(true)
+        })
+    }
+
+    const handleDislike = (event) => {
+        event.preventDefault()
+        const body = {
+            inc_votes : -1,
+        }
+        setIncrement(-1)
```

Comment:
rather than having this additional increment state, you can directly update the votes in state - as you are currently doing this optimistically, you will need to undo this in the catch block in case of an error